pymc3/models/base.py

```diff
@@ -0,0 +1,69 @@
+from ..model import modelcontext
+from collections import OrderedDict
+
+
+class UserModel(object):
+    """Base class for model specification
+    It is supposed to be used by pymc3 contributors
+    for simplifying usage of common bayesian models
+    """
+
+    def __init__(self, name):
+        # name should be used as prefix for
+        # all variables specified within a model
+        self.vars = OrderedDict()
+        self.name = name
+
+    @property
+    def model(self):
+        """Shortcut to model"""
+        return modelcontext(None)
+
+    def new_var(self, name, dist, data=None, test_val=None):
```

Comment:
I can see how this is a useful abstraction but it makes transcribing a PyMC3 model from the usual syntax to be a class a bit harder as it gives cognitive overhead. After all, the code below isn't doing so much that it warrants the indirection. Is there another way we can achieve that?